ReactiveCocoa/Swift/Property.swift

```diff
@@ -154,16 +208,17 @@ public final class MutableProperty<Value>: MutablePropertyType {
 	/// By definition, this only works if the object given to init() is
 	/// KVO-compliant. Most UI controls are not!
 	public var producer: SignalProducer<AnyObject?, NoError> {
-		if let object = object {
-			return object.rac_valuesForKeyPath(keyPath, observer: nil).toSignalProducer()
-				// Errors aren't possible, but the compiler doesn't know that.
-				.flatMapError { error in
-					0 // suppresses implicit return error on fatalError
-					fatalError("Received unexpected error from KVO signal: \(error)")
-				}
-		} else {
-			return .empty
+		return _property?.producer ?? .empty
+	}
+
+	public var signal: Signal<AnyObject?, NoError> {
+		guard let _property = _property else {
```

Comment:
It is because the DynamicProperty can possibly be retained and outlive the underlying object. So in order to have the signal and producer complete at the time the underlying object is gone, the underlying MutableProperty must be deallocated.